core/src/AzureMcp.Cli/AzureMcp.Cli.csproj

```diff
@@ -13,18 +13,28 @@
     <IncludeNativeLibrariesForSelfExtract>true</IncludeNativeLibrariesForSelfExtract>
     <DebugType>embedded</DebugType>
   </PropertyGroup>
-  <!-- AOT compilation flags -->
+
+  <!-- Enable Trim and AOT Analyzers -->
   <PropertyGroup>
-    <IlcFoldIdenticalMethodBodies>true</IlcFoldIdenticalMethodBodies>
-    
-  <!-- Enable Analyzers -->
     <EnableAotAnalyzer>true</EnableAotAnalyzer>
     <EnableTrimAnalyzer>true</EnableTrimAnalyzer>
     <EnableSingleFileAnalyzer>true</EnableSingleFileAnalyzer>
     <TrimmerSingleWarn>false</TrimmerSingleWarn>
     <SuppressTrimAnalysisWarnings>false</SuppressTrimAnalysisWarnings>
   </PropertyGroup>
 
+  <!-- AOT native build flags -->
+  <PropertyGroup Condition="'$(BuildNative)' == 'true'">
+    <IsAotCompatible>true</IsAotCompatible>
+    <PublishAot>true</PublishAot>
+    <PublishTrimmed>true</PublishTrimmed>
+    <TrimMode>full</TrimMode>
+    <SelfContained>true</SelfContained>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```

Comment:
I think this setting is agnostic of aoe building or not... may want to promote this to top level property group

Reply:
make sense, moved `TreatWarningsAsErrors` to top level property group